src/Fmt.ml

```diff
@@ -164,11 +164,14 @@ let box_depth_colors = [|32; 33; 94; 31; 35; 36|]
 let box_depth_color () =
   box_depth_colors.(!box_depth % Array.length box_depth_colors)
 
-let debug_box_open box_kind n fs =
+let debug_box_open ?name box_kind n fs =
   if !box_debug_enabled then (
-    let openning =
-      if n = 0 then box_kind else Format.sprintf "%s<%d" box_kind n
+    let name =
+      Option.value_map name
+        ~f:(fun s -> Format.sprintf "%s:%s" box_kind s)
+        ~default:box_kind
```

Comment:
Nitpick: I think using a match instead of `Option.value_map` is both easier to type and to read.

Reply:
done